### PR TITLE
add scalar tensor on spyre

### PR DESCRIPTION
**Pull request by thoangtrvn**
Thanks for sending a pull request! Please make sure you read the [contributing guidelines](https://github.com/torch-spyre/torch-spyre/blob/main/CONTRIBUTING.md).

#### What type of PR is this?

- [ ] bug
- [x] feature
- [ ] documentation
- [ ] cleanup

#### What this PR does:

This PR now allows creating a scalar tensor on Spyre. 

#### Additional note:

CC: @joyalbin

### CODE REVIEW

File: tests/inductor/test_inductor_ops.py [hunk 1]
```diff
@@ -297,7 +297,7 @@ class TestOps(unittest.TestCase, metaclass=ParameterizedTestMeta):
                 "3d_dim_neg1": (-1, cached_randn((3, 7, 9))),
                 "3d_dim_neg12": ((-1, -2), cached_randn((3, 7, 9))),
                 # 0D / scalar tensor:
-                "scalar_tensor": (None, torch.tensor(5.0, dtype=torch.float16)),
+                # "scalar_tensor": (None, torch.tensor(5.0, dtype=torch.float16)),
```
Comment (ani300) [inline tests/inductor/test_inductor_ops.py]: this test should work right?

**Comment by joyalbin** [comment]
@thoangtrvn 
Is there any plan to support 0D at kernel launch also?
scalar tensor 0D is converted to 1D at https://github.com/torch-spyre/torch-spyre/blob/main/torch_spyre/csrc/module.cpp#L183

**Comment by thoangtrvn** [comment]
> @thoangtrvn Is there any plan to support 0D at kernel launch also? scalar tensor 0D is converted to 1D at https://github.com/torch-spyre/torch-spyre/blob/main/torch_spyre/csrc/module.cpp#L183

@joyalbin : yes, this PR allows creating 0D tensor on spyre. There is a need to handle 0D tensor in certain ops that go through torch.compile() for spyre - an example is the amax that I currently have to disable the test. At the kernel launch level, we will replace this graph-based kernel launch which then we support natively 0D tensor, as long as the torch.compile() for that op doesn't give any error.

**Comment by yoheiueda** [comment]
I think issues related to 0D support for individual operators can be handled incrementally as part of the op‑enablement work.

@thoangtrvn Would it make sense to start with a PR that enables 0D support in the kernel‑launch code? If some test cases start failing as a result, it seems reasonable to temporarily disable those tests. We can then follow up by fixing the issues on a per‑operator basis.

**Comment by thoangtrvn** [comment]
> I think issues related to 0D support for individual operators can be handled incrementally as part of the op‑enablement work.
> 
> @thoangtrvn Would it make sense to start with a PR that enables 0D support in the kernel‑launch code? If some test cases start failing as a result, it seems reasonable to temporarily disable those tests. We can then follow up by fixing the issues on a per‑operator basis.

@ani300 : are you ok with this?

**Review by ani300 — DISMISSED** [review]
PR looks good to me, especially if it fixes a lot of 0D tensor errors we've been hitting with model enablement

**Comment by thoangtrvn** [comment]
@ani300 , @avery-blanchard , @yoheiueda : the scalar tensor support should be ready. Summary:
* add one new test `test_scalar_tensor` to check dim() == 0 on scalar tensor created on spyre. 
* remove expect_fail from 2 tests in test_spyre.py, addressing issue https://github.com/torch-spyre/torch-spyre/issues/1172
* enable 'scalar_tensor' tests in test_inductor_ops.py - no longer expected fail.  
* fully support 0D tensors on spyre. 

CC: @JRosenkranz

**Review by ani300 — APPROVED** [review]
lgtm! that should unblock a bunch of model-enablement issues

**Review by JRosenkranz — APPROVED** [review]
lgtm

**Comment by thoangtrvn** [comment]
We need to wait for the fix in deeptools to be in CI image for the PR to pass the tests. 

MAIN:
image

THIS PR:
image

**Review by dgrove-oss — APPROVED** [review]
lgtm
